components/suggest/src/schema.rs

```diff
@@ -121,6 +121,10 @@ pub const SQL: &str = "
         description TEXT NOT NULL,
         FOREIGN KEY(suggestion_id) REFERENCES suggestions(id) ON DELETE CASCADE
     );
+
+    CREATE TABLE dismissed_suggestions (
+        url TEXT PRIMARY KEY
```

Comment:
Directly indexing URLs might lead to sub-optimal query performance as vast majority of them begin with "https://www." especially against large tables. We could either index the reversed URLs or follow Places' wisdom of indexing on URL hashes instead.

```
CREATE TABLE dismissed_urls (
    url_hash INTEGER NOT NULL,
    url TEXT NOT NULL,
) WITHOUT ROWID;

CREATE INDEX idx_dismissed_urls_url_hash ON dismissed_urls (url_hash);

SELECT
  1
FROM
  dismissed_urls
WHERE
  url_hash = MD5(target_url)  -- Much faster index lookup
AND
  url = target_url  -- Needed to avoid hash collisions
```

Alternatively, for dismissal records, we often only store the URL hashes in the browser as the collision rate should be super low, and the collision impact (i.e. not serving a suggestion) is normally acceptable.

What do you think?

Reply:
That makes sense to me.  I switched to just storing the hash.